sbc_translation/models/correspondence.py

```diff
@@ -22,22 +33,119 @@ class Correspondence(models.Model):
+        if smb_conn.connect(smb_ip, smb_port):
+            file_ = BytesIO(base64.b64decode(
+                letter.letter_image.with_context(
+                    bin_size=False).datas))
+            config_obj = self.env['ir.config_parameter']
+            nas_share_name = (config_obj.search(
+                [('key', '=', 'sbc_translation.nas_share_name')])[0]).value
+
+            child = sponsorship.child_id
+            sponsor = sponsorship.partner_id
+            letter_name = "_".join(
+                (child.code, sponsor.ref, str(letter.id)))
+            letter_name = ".".join((letter_name, 'pdf'))
+
+            nas_letters_store_path = (config_obj.search(
+                [('key', '=', 'sbc_translation.nas_letters_store_path')])
+                [0]).value + letter_name
+            smb_conn.storeFile(nas_share_name,
+                               nas_letters_store_path, file_)
+
+            logger.info('File {} store on NAS with success'
+                        .format(letter.letter_image.name))
+
+            letter.state = 'Global Partner translation queue'
+        else:
+            raise Exception('Connection to NAS failed')
+
+    # CRON Methods
+    ##############
+    @api.model
+    def check_local_translation_done(self):
+        tc = translate_connector.TranslateConnect()
+        letters_to_update = tc.get_translated_letters()
+
+        if not letters_to_update == -1:
+            for letter in letters_to_update:
+                correspondence = self.browse(letter["letter_odoo_id"])
+                reload(sys)
+                sys.setdefaultencoding('UTF8')
+
+                tg_lang = letter["target_lang"]
+                target_lang_id = self.env['res.lang.compassion'].search(
+                    [('code_iso', '=', tg_lang)]).id
+                # find the good text to writte
+                if tg_lang == 'eng':
+                    target_text = 'english_text'
+                elif tg_lang in self.get_child_langs_code(correspondence):
+                    target_text = 'translated_text'
+                else:
+                    logger.error("Correspondence letters language no match")
+
+                # UPDATE Odoo Database
+                correspondence.write(
+                    {target_text: letter["text"],
+                     'state': 'Received in the system',
+                     'destination_language_id': target_lang_id})
+
+                # Send to GMC
+                if correspondence.direction == 'Supporter To Beneficiary':
+                    action_id = self.env.ref(
+                        'onramp_compassion.create_commkit').id
+                    self.env['gmc.message.pool'].create({
+                        'action_id': action_id,
+                        'object_id': letter["letter_odoo_id"]
+                    })
+
+                tc.remove_letter(letter["id"])
+
+    def get_child_langs_code(self, sponsorship):
+        """ Return children's code iso """
+        codes = []
```

Comment:
I think you don't need a method for that as you can replace by:
`sponsorship.child_id.project_id.country_id.spoken_lang_ids.mapped('code_iso')`